include/chaiscript/language/chaiscript_engine.hpp

```diff
@@ -204,6 +204,27 @@ namespace chaiscript
       m_engine.add(fun([this](const std::string& t_namespace_name) { import(t_namespace_name); }), "import");
     }
 
+    /// Skip BOM at the beginning of file
+    static bool skip_bom(std::ifstream &infile) {
+        size_t bytes_needed = 3;
+        std::vector<char> v(bytes_needed);
+
+        infile.read(&v[0], static_cast<std::streamsize>(bytes_needed));
+        std::string buffer_string(v.begin(), v.end());
+
+        if (!infile.eof()
```

Comment:
The more I think about this check, the less sense it makes. Right now the BOM wouldn't be skipped if the file includes only the BOM (I think; not 100% sure if the eof bit is set already).

Either way, I think the better approach would be to allocate something like `char header[3] = {0};`, read the first 3 bytes from the stream and just compare them 1:1 with the BOM sequence, not even testing the length/position (since failed reads would result in `\0`).